src/docfx/lib/log/Telemetry.cs

```diff
@@ -63,8 +64,9 @@ public static void TrackException(Exception ex)
 
         public static void Flush()
         {
-            // Default timeout of 100 sec is used
-            s_telemetryClient.Flush();
+            // Default timeout of TelemetryClient.Flush is 100 seconds,
+            // but we only want to wait for 2 seconds at most.
+            Task.WaitAny(Task.Run(s_telemetryClient.Flush), Task.Delay(2000));
```

Comment:
what will happen if it's time out?

Reply:
nothing, just hand there doing nothing for 100s

Reply:
https://github.com/Microsoft/ApplicationInsights-dotnet/issues/407

Reply:
will we lost any traces here if it's timeout?

Reply:
Nice catch, @OsmondJiang ! I just recalled in the legacy `TraceEx` library, there is also a workaround to sleep for 10s.

Reply:
@OsmondJiang @herohua Flush is sync when using `InMemoryChannel`, which is the default channel used in docfx, so we don't have the problem mentioned in that issue. This change is just to [give Flush a timeout](https://github.com/Microsoft/ApplicationInsights-dotnet/issues/1088)